atomate/vasp/firetasks/parse_outputs.py

```diff
@@ -775,6 +776,652 @@ def run_task(self, fw_spec):
         # a builder to put it into materials collection... -computron
         logger.info("Thermal expansion coefficient calculation complete.")
 
+@explicit_serialize
+class HubbardHundLinRespToDb(FiretaskBase):
```

Comment:
The code quality looks good here, but it is very difficult to follow along because there is so much code in a single function.

Can you separate out the internal functions and put them in their own file (e..g, `atomate/vasp/analysis/linear_response.py`) and make sure that they have their own docstrings.

Furthermore, can you create new functions for distinct parts of the code. E.g., if you have a lot of code inside a loop, you can move the code inside the loop to a new function and just call the function in the loop.

In general, each function shouldn't be more than say 50 lines long otherwise they get unwieldy.